turbo/stages/bodydownload/body_algos.go

```diff
@@ -76,53 +75,77 @@ func (bd *BodyDownload) RequestMoreBodies(tx kv.RwTx, blockReader services.FullB
 	var bodyReq *BodyRequest
 	blockNums := make([]uint64, 0, BlockBufferSize)
 	hashes := make([]common.Hash, 0, BlockBufferSize)
-	for ; len(blockNums) < BlockBufferSize && bd.requestedLow <= bd.maxProgress; blockNum++ {
+
+	// go right back to the start of the earliest request that it still outstanding here so that we can
+	// check for timeouts, as requests are delivered from peers bd.requestedLow will be incremented so this loop will move forwards over time.
+	// If a request has timed out we nil out the request and add it back in to the next request going to a peer.
+	// not all requests will have contiguous block numbers because of this
+	for bNum := bd.requestedLow; len(blockNums) < BlockBufferSize && bNum <= bd.maxProgress; bNum++ {
```

Comment:
The intent is not to always start from `bd.requestedLow`, but from it was left off the last time, `blockNum` (which is both the argument and return value)

Reply:
Actually, you explained it in the comments to the PR... in which case perhaps `blockNum` can be removed?

Reply:
Absolutely!  I've removed this now